http-client/src/client.rs

```diff
@@ -1,11 +1,38 @@
 use crate::transport::HttpTransportClient;
 use async_trait::async_trait;
 use jsonrpc::DeserializeOwned;
-use jsonrpsee_types::{error::Error, http::HttpConfig, jsonrpc, traits::Client};
+use jsonrpsee_types::{error::Error, jsonrpc, traits::Client};
 
 use std::convert::TryInto;
 use std::sync::atomic::{AtomicU64, Ordering};
 
+/// Http Client Builder.
```

Comment:
removed `types/http` types and introduces builder patterns for the `HTTP server and client`